--- a/gitstats	2024-06-24 21:09:20.426186008 +0200
+++ b/gitstats	2024-06-25 09:12:22.390452602 +0200
@@ -96,7 +96,7 @@
 	return [el[1] for el in sorted([(d[el][key], el) for el in list(d.keys())])]
 
 def getstatsummarycounts(line):
-	numbers = re.findall('\d+', line)
+	numbers = re.findall(r'\d+', line)
 	if   len(numbers) == 1:
 		# neither insertions nor deletions: may probably only happen for "0 files changed"
 		numbers.append(0);
@@ -322,7 +322,7 @@
 				continue
 			prev = tag
 			for line in output.split('\n'):
-				parts = re.split('\s+', line, 2)
+				parts = re.split(r'\s+', line, 2)
 				commits = int(parts[1])
 				author = parts[2]
 				self.tags[tag]['commits'] += commits
@@ -485,7 +485,7 @@
 		for line in lines:
 			if len(line) == 0:
 				continue
-			parts = re.split('\s+', line, 4)
+			parts = re.split(r'\s+', line, 4)
 			if parts[0] == '160000' and parts[3] == '-':
 				# skip submodules
 				continue
