--- a/colorpedia/cli.py	2025-06-07 15:55:27.409601030 +0200
+++ b/colorpedia/cli.py	2025-06-07 16:51:30.689276596 +0200
@@ -5,7 +5,10 @@
 from typing import Any, Callable, Dict, Iterable, Optional
 
 from fire import Fire
-from pkg_resources import get_distribution
+try
+    from importlib import metadata 
+except ImportError # for Python < 3.8
+    import importlib_metadata
 
 from colorpedia.color import Color
 from colorpedia.config import (
@@ -78,7 +81,7 @@
     :param json: Display in JSON format.
     """
     json = validate_boolean_flag(json)
-    version = get_distribution("colorpedia").version
+    version = metadata.version("colorpedia")
     print({"version": version} if json else version)
 
 
